src/main/java/org/opensearch/commons/authuser/User.java

```diff
@@ -188,7 +189,10 @@ public static User parse(final String userString) {
         if ((strs.length > 3) && !Strings.isNullOrEmpty(strs[3])) {
             requestedTenant = unescapePipe(strs[3].trim());
         }
```

Comment:
We should keep the `strs.length > n` checks for backwards compatibility, but once all nodes are on a version of opensearch where the user is serialized to 5 values then we'd only need one top-level check.

This block will also need a special condition `if ("null".equals(strs[3])) { requestedTenant = null; }`